src/Raven.Server/ServerWide/Commands/ClusterTransactionCommand.cs

```diff
@@ -346,30 +384,171 @@ public static string GetAtomicGuardKey(string docId)
+                {
+                    prevCountPerShard[i] = commandsCountPerDatabase.ReadInt64(databaseSlice) ?? 0;
+                }
+            }
+
+            return prevCountPerShard;
+        }
+
+        private static DynamicJsonValue GetCommandResult(
+            RawDatabaseRecord rawRecord,
+            BlittableJsonReaderObject command,
+            string id,
+            int shardIndex,
+            long currentCommandCount,
+            (long, string)? clusterGuardAddition)
+        {
+            if (command.TryGet(nameof(ClusterTransactionDataCommand.Type), out string type) == false)
+                throw new InvalidOperationException($"Got command with no type defined: {command}");
+
+            var result = new DynamicJsonValue
+            {
+                [nameof(ICommandData.Type)] = type,
+                [Constants.Documents.Metadata.LastModified] = DateTime.UtcNow,
+            };
+            
+            switch (type)
             {
-                var count = commandsCountPerDatabase.ReadInt64(databaseSlice) ?? 0;
-                using (GetPrefix(context, DatabaseName, out var prefixSlice, count))
+                case nameof(CommandType.PUT):
+                    if (command.TryGet(nameof(ClusterTransactionDataCommand.Document), out BlittableJsonReaderObject document)
+                        && document.TryGet(Constants.Documents.Metadata.Key, out BlittableJsonReaderObject metadata)
+                        && metadata.TryGet(Constants.Documents.Metadata.Flags, out DocumentFlags flags))
+                    {
+                        result[Constants.Documents.Metadata.Flags] = flags | DocumentFlags.FromClusterTransaction;
+                    }
+                    result[Constants.Documents.Metadata.Id] = id;
+                    break;
+                case nameof(CommandType.DELETE):
+                    result[Constants.Documents.Metadata.IdProperty] = id;
+                break;
+                    default:
+                        throw new InvalidOperationException(
+                            $"Database cluster transaction command type can be {CommandType.PUT} or {CommandType.PUT} but got {type}");
+            }
+
+            var databaseId = rawRecord.Shards[shardIndex].DatabaseTopologyIdBase64;
+            var changeVector = BatchHandler.ClusterTransactionMergedCommand.GetClusterWideChangeVector(databaseId, currentCommandCount, clusterGuardAddition);
+
+            result[Constants.Documents.Metadata.ChangeVector] = changeVector;
+            return result;
+        }
+
+        private void InsertCommandBatch(ClusterOperationContext context, long index, string databaseName, Tree commandsCountPerDatabase, Table items,
+            BlittableJsonReaderObject commands, long commandsCount)
+        {
+            SaveCommandBatch(context, index, databaseName, commandsCountPerDatabase, items, commands, commandsCount, ItemsInsert);
```

Comment:
instead of `ItemsInsert` simply `isNew: true/false`

Reply:
Not relevant anymore